packages/gestalt/src/Toast.js

```diff
@@ -44,15 +68,14 @@ type Props = {|
   /**
    * An optional thumbnail image to displayed next to the text.
    */
-  thumbnail?: Node,
-  /**
-   * The masked shape of the thumbnail.
-   */
-  thumbnailShape?: 'circle' | 'square',
+  thumbnail?:
```

Comment:
Using an object here seems a bit belt-and-suspenders given the `displayName` checks. I think we're fine typing this as a simple union of the acceptable components and using `displayName` where we're checking the object key. (Unless I'm missing some wacky usage — I don't think there's anything like "SterlingAvatar" that people use, is there?)

Reply:
@dangerismycat The only caveat us that we are cloning them to override sizes. I think this is fine, was just curious if we thought maybe objects could be more "neat".

Reply:
Hm…I'm not sure I see the need for the object? I think I'm missing something though. Let's discuss in next Tuesday's eng meeting? This is an interesting topic

Reply:
Keeping as is as per https://flow.org/en/docs/types/unions/#toc-disjoint-object-unions

Reply:
After chatting with Jack

Reply:
we could use `thumbnail instanceof "Image"` to narrow type and it works. 

With Flow better not do much polymorphism with objects, usually disjoint object unions are recommended

Reply:
DONE

Reply:
See https://github.com/pinterest/gestalt/pull/2525#issuecomment-1462300583

Reply:
Well dang. At least we tried! Thank you for documenting how each approach didn't work, much appreciated both for now and for the future.